src/test/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacadeTest.java

```diff
@@ -0,0 +1,111 @@
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+public class FlashcardFacadeTest extends FlashcardAndQuestionTagAbstractTests {
+    @Autowired
+    private FlashcardFacade flashcardFacade;
+    @MockBean
+    private FlashcardService flashcardService;
+    @MockBean
+    private QuestionTagService questionTagService;
+    private Flashcard flashcard;
+
+    @Before
+    public void createFlashcard() {
+        flashcard = new Flashcard();
+        flashcard.setId(1L);
+        flashcard.setQuestion("question");
+        flashcard.setAnswer("answer");
+        flashcard.setExplanation("explanation");
+        //questionTag with flashcards
+        QuestionTag questionTag = new QuestionTag();
+        questionTag.setName("home");
+        questionTag.setFlashcards(new HashSet<>(Collections.singletonList(new Flashcard("kubek", "mug"))));
+        flashcard.getQuestionTagsSet().add(questionTag);
+    }
+
+    @Test
+    public void saveFlashcard_ShouldReturnFlashcardSaveResponseDto() {
+        //given
+        FlashcardSaveDto flashcardSaveDto = new FlashcardSaveDto();
+
+        Set<QuestionTag> questionTagSet = new HashSet<>(Arrays.asList(createQuestionTag(1L, "home"),
+                createQuestionTag(2L, "holiday")));
+        flashcardSaveDto.setTagsSet(new HashSet<>(Arrays.asList("home", "holiday")));
+
+        when(flashcardService.saveFlashcard(any(Flashcard.class)))
+                .thenReturn(flashcard);
+        when(questionTagService.getQuestionTagsSet(anySet()))
+                .thenReturn(questionTagSet);
+
+        Long expectedId = 1L;
+        //when
+        FlashcardSaveResponseDto flashcardSaveResponseDto = flashcardFacade.saveFlashcard(flashcardSaveDto);
+        List<QuestionTag> questionTagList = new ArrayList<>(questionTagSet);
+        //then
+        Assert.assertNotNull(flashcardSaveResponseDto);
+        Assert.assertEquals(expectedId, flashcardSaveResponseDto.getId());
+        Assert.assertTrue(questionTagList.get(0).getFlashcards().contains(flashcard));
+        Assert.assertTrue(questionTagList.get(1).getFlashcards().contains(flashcard));
+    }
+
+    @Test
+    public void whenUpdateFlashcard_ThenShouldInvokeDeleteUselessQuestionTagsInQuestionTagService_AndReturnExpectedFlashcardSaveResponseDto() {
+        //given
+        when(flashcardService.findById(anyLong()))
+                .thenReturn(flashcard);
+        when(questionTagService.getQuestionTagsSet(any()))
+                //question tag to add
+                .thenReturn(new HashSet<>(Collections.singletonList(new QuestionTag("holiday"))))
+                //question tag to remove
+                .thenReturn(new HashSet<>(Collections.singletonList(new QuestionTag("home"))));
+        //when
+        FlashcardSaveResponseDto flashcardSaveResponseDto = flashcardFacade.updateFlashcard(super.flashcardSaveDto, 1L);
+        //then
+        Assert.assertSame(1L, flashcardSaveResponseDto.getId());
```

Comment:
1L is default value, it's already set. Maybe let's use different value to see if it's really updated.

Reply:
I don't update flashcard id in updateFlashcard